desktop-widgets/modeldelegates.cpp

```diff
@@ -254,9 +255,16 @@ TankUseDelegate::TankUseDelegate(QObject *parent) : QStyledItemDelegate(parent)
 
 QWidget *TankUseDelegate::createEditor(QWidget *parent, const QStyleOptionViewItem &, const QModelIndex &) const
 {
+	struct divecomputer *currentDc;
+	if (DivePlannerPointsModel::instance()->currentMode() != DivePlannerPointsModel::NOTHING)
+		currentDc = &displayed_dive.dc;
+	else
+		currentDc = get_dive_dc(current_dive, dc_number);
+	bool isCcrDive = currentDc ? currentDc->divemode == CCR : true;
 	QComboBox *comboBox = new QComboBox(parent);
 	for (int i = 0; i < NUM_GAS_USE; i++)
-		comboBox->addItem(gettextFromC::tr(cylinderuse_text[i]));
+		if (isCcrDive || ((i != DILUENT) && (i != OXYGEN)))
```

Comment:
Coding style: add braces to the `for`-loop since the body grew beyond a single line.

Also I am not sure if the parentheses around `i != DILUENT` and `i != OXYGEN` are idiomatic.